### PR TITLE
Rename test app packages

### PR DESCRIPTION
**Pull request by kevinlind**
## Description

Applies some fixes to the Java and Kotlin test apps:

- Move Java test app package to `com.adobe.edge.media.testapp`
- Move Kotlin test app package to `com.adobe.edge.media.testappkotlin`
- Add deep link support for Assurance using `testapp://main`
- Add extension version to TextView in main activity
- Update getting started with test app with Assurance setup instructions.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: code/testapp/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-    package="com.adobe.mediaanalyticstestapp">
+    package="com.adobe.edge.media.testapp">
 
     <uses-permission android:name="android.permission.INTERNET" />
     <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
 
     <application
         android:networkSecurityConfig="@xml/network_security_config"
-        android:name=".MediaAnalyticsTestApp"
+        android:name=".MediaTestApp"
```
Comment (addb) [inline code/testapp/src/main/AndroidManifest.xml:10]: nit: Should we name it EdgeMediaTestApp

File: code/testapp/src/main/java/com/adobe/edge/media/testapp/MediaTestApp.java [hunk 2]
```diff
@@ -21,7 +21,7 @@
 import com.adobe.marketing.mobile.edge.media.Media;
 import java.util.Arrays;
 
-public class MediaAnalyticsTestApp extends Application {
+public class MediaTestApp extends Application {
```
Comment (addb) [inline code/testapp/src/main/java/com/adobe/edge/media/testapp/MediaTestApp.java:24]: nit: `TestApp` generic name

File: code/testapp/src/main/java/com/adobe/edge/media/testapp/analytics/MediaAnalyticsProvider.java [hunk 3]
```diff
@@ -9,15 +9,19 @@
   governing permissions and limitations under the License.
 */
 
-package com.adobe.mediaanalyticstestapp.analytics;
+package com.adobe.edge.media.testapp.analytics;
```
Comment (addb) [inline code/testapp/src/main/java/com/adobe/edge/media/testapp/analytics/MediaAnalyticsProvider.java:12]: let's rename analytics package to either 
`tracking` || `handler` || `api.implementation`

Comment (addb) [reply code/testapp/src/main/java/com/adobe/edge/media/testapp/analytics/MediaAnalyticsProvider.java:12]: Let's rename the file to 
EdgeMediaAPIImplementation
EdgeMediaAPICaller
EdgeMediaTracker

File: code/testapp/src/main/AndroidManifest.xml (hunk 1 above)
Comment (kevinlind) [reply code/testapp/src/main/AndroidManifest.xml:10]: This name needs to match the Application class name.

File: code/testapp/src/main/java/com/adobe/edge/media/testapp/MediaTestApp.java (hunk 2 above)
Comment (kevinlind) [reply code/testapp/src/main/java/com/adobe/edge/media/testapp/MediaTestApp.java:24]: In this case I don't think we need to make the application class name generic. The Edge test app does use `TestApplication` but for Consent we use `ConsentTestApplication` and for Identity it's `EdgeIdentityApplication`. Unless you feel strongly to change the name to something like `EdgeMediaTestApplication`, I'm inclined to leave the class name as is.

File: code/testapp/src/main/java/com/adobe/edge/media/testapp/analytics/MediaAnalyticsProvider.java (hunk 3 above)
Comment (kevinlind) [reply code/testapp/src/main/java/com/adobe/edge/media/testapp/analytics/MediaAnalyticsProvider.java:12]: What about renaming the package to `tracker` and the class to `MediaPlayerObserver` as the class implements the `Observer` interface?

Comment (addb) [reply code/testapp/src/main/java/com/adobe/edge/media/testapp/analytics/MediaAnalyticsProvider.java:12]: Sounds good to me @kevinlind.

**Review by addb — APPROVED** [review]

**Pull request merged**